src/Illuminate/Support/Collection.php

```diff
@@ -296,6 +296,34 @@ public function containsStrict($key, $value = null)
         return in_array($key, $this->items, true);
     }
 
+    /**
+     * Join all items from the collection using a string. The final items can use a seperate glue string.
+     *
+     * @param  string  $glue
+     * @param  string  $finalGlue
+     * @return string
+     */
+    public function join($glue, $finalGlue = '')
+    {
+        if ($finalGlue === '') {
```

Comment:
We could use `if ($finalGlue === '' || $this->count() < 2) {` and remove the other two `if` clauses.

Reply:
I think having separate `if`s is more readable, but we'll see what @taylorotwell decides on.